src/assignthree/model/Graph.java

```diff
@@ -0,0 +1,198 @@
+package assignthree.model;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import assignthree.util.DfsOperatorUtils;
+import assignthree.exceptions.InvalidInput;
+
+/*
+ * Class having all functions related to tree
+ */
+public class Graph {
+
+    /*
+     * dependency of from parent to child relations
+     */
+    private Map<Integer, ArrayList<Integer>> child = new ConcurrentHashMap<>();
```

Comment:
memory wastage. try to put it in get function in case the variable is null

Reply:
ok